src/Datagrid/ProxyQuery.php

```diff
@@ -202,22 +201,8 @@ public function execute(array $params = [], $hydrationMode = null)
         }
 
         $query = $this->getDoctrineQuery();
-        $queryBuilder = $this->getQueryBuilder();
-
-        $identifierFieldNames = $queryBuilder
-            ->getEntityManager()
-            ->getMetadataFactory()
-            ->getMetadataFor(current($this->getQueryBuilder()->getRootEntities()))
-            ->getIdentifierFieldNames();
-        $hasSingleIdentifierName = 1 === \count($identifierFieldNames);
-        $hasJoins = 0 !== \count($queryBuilder->getDQLPart('join'));
-        $havingPart = $queryBuilder->getDQLPart('having');
-        $hasHavingPart = \is_array($havingPart) && (\count($havingPart) > 0);
-
-        if (!$hasJoins) {
-            $query->setHint(CountWalker::HINT_DISTINCT, false);
-        }
 
+        // NEXT_MAJOR: Remove this block.
```

Comment:
Why this block can't be remove now since it's in the Paginator ?

Reply:
I left it because there is this code after:

```php
// NEXT_MAJOR: Remove this.
if (\func_num_args() > 0) {
    return $query->execute($params, $hydrationMode);
}
```
So it could break BC.